### ISSUE TITLE
Build broken in glibc-2.40.9000-409-g776938e8b8.tar.xz

### ISSUE THREAD

**Issue by johnny-mnemonic** [linux-ia64/glibc-ia64#3]
Trying to cross-build https://snapshots.sourceware.org/glibc/trunk/latest/src/glibc-2.40.9000-409-g776938e8b8.tar.xz in T2 with https://github.com/linux-ia64/glibc-ia64/compare/master...master-epic ([master-epic@39a08f8](https://github.com/linux-ia64/glibc-ia64/tree/39a08f8aa5861c0b7396aa0daea8ea4c59b2c25f)) on top fails and yields:

```
[...]
/usr/src/t2-src/build/ia64-toolchain-24-svn-generic-ia64-itanium2-cross-linux/TOOLCHAIN/cross/bin/../lib/gcc/ia64-t2-linux-gnu/15.0.0/../../../../ia64-t2-linux-gnu/bin/ld: warning: -z relro ignored
/usr/src/t2-src/build/ia64-toolchain-24-svn-generic-ia64-itanium2-cross-linux/TOOLCHAIN/cross/bin/../lib/gcc/ia64-t2-linux-gnu/15.0.0/../../../../ia64-t2-linux-gnu/bin/ld: /usr/src/t2-src/src.glibc.ia64-toolchain.241205.214533.115695/glibc-2.40.9000-419-gdae2e746b7/objs/math/libm_pic.a(s_cospil.os): in function `__cospil':
s_cospil.c:(.text+0x142): undefined reference to `__roundl'
/usr/src/t2-src/build/ia64-toolchain-24-svn-generic-ia64-itanium2-cross-linux/TOOLCHAIN/cross/bin/../lib/gcc/ia64-t2-linux-gnu/15.0.0/../../../../ia64-t2-linux-gnu/bin/ld: /usr/src/t2-src/src.glibc.ia64-toolchain.241205.214533.115695/glibc-2.40.9000-419-gdae2e746b7/objs/math/libm_pic.a(s_sinpil.os): in function `__sinpil':
s_sinpil.c:(.text+0x152): undefined reference to `__roundl'
/usr/src/t2-src/build/ia64-toolchain-24-svn-generic-ia64-itanium2-cross-linux/TOOLCHAIN/cross/bin/../lib/gcc/ia64-t2-linux-gnu/15.0.0/../../../../ia64-t2-linux-gnu/bin/ld: /usr/src/t2-src/src.glibc.ia64-toolchain.241205.214533.115695/glibc-2.40.9000-419-gdae2e746b7/objs/math/libm_pic.a(s_cospi.os): in function `__cospi':
s_cospi.c:(.text+0x122): undefined reference to `__round'
/usr/src/t2-src/build/ia64-toolchain-24-svn-generic-ia64-itanium2-cross-linux/TOOLCHAIN/cross/bin/../lib/gcc/ia64-t2-linux-gnu/15.0.0/../../../../ia64-t2-linux-gnu/bin/ld: /usr/src/t2-src/src.glibc.ia64-toolchain.241205.214533.115695/glibc-2.40.9000-419-gdae2e746b7/objs/math/libm_pic.a(s_sinpi.os): in function `__sinpi':
s_sinpi.c:(.text+0x142): undefined reference to `__round'
collect2: error: ld returned 1 exit status
make[2]: *** [../Makerules:535: /usr/src/t2-src/src.glibc.ia64-toolchain.241205.214533.115695/glibc-2.40.9000-419-gdae2e746b7/objs/math/libm.so] Error 1
make[2]: Leaving directory '/usr/src/t2-src/src.glibc.ia64-toolchain.241205.214533.115695/glibc-2.40.9000-419-gdae2e746b7/math'
make[1]: *** [Makefile:484: math/others] Error 2
make[1]: Leaving directory '/usr/src/t2-src/src.glibc.ia64-toolchain.241205.214533.115695/glibc-2.40.9000-419-gdae2e746b7'
make: *** [Makefile:9: all] Error 2
[...]
```

See https://github.com/johnny-mnemonic/toolchain-autobuilds/actions/runs/12172730052/job/33951924167#step:4:118 and corresponding [build logs](https://github.com/johnny-mnemonic/toolchain-autobuilds/actions/runs/12172730052/artifacts/2277286073). Also attached here:  [build-logs.zip](https://github.com/user-attachments/files/18030834/build-logs.zip)

**Comment by johnny-mnemonic** [comment]
This is related to the recent addition of C23 functionality with:
* [Implement C23 cospi](https://sourceware.org/git/?p=glibc.git;a=commit;h=0ae0af68d8fa3bf6cbe1e4f1de5929ff71de67b3)
* [Implement C23 sinpi](https://sourceware.org/git/?p=glibc.git;a=commit;h=776938e8b8dcf2b59998979e91cc0f9db7d771a8)

...and can be avoided by:

1. either dropping the architecture specific implementations for round* functions with e.g.:
```diff
diff --git a/sysdeps/ia64/math-use-builtins-round.h b/sysdeps/ia64/math-use-builtins-round.h
new file mode 100644
index 0000000000..6ed77781ad
--- /dev/null
+++ b/sysdeps/ia64/math-use-builtins-round.h
@@ -0,0 +1,4 @@
+#define USE_ROUND_BUILTIN 1
+#define USE_ROUNDF_BUILTIN 1
+#define USE_ROUNDL_BUILTIN 1
+#define USE_ROUNDF128_BUILTIN 0
\ No newline at end of file
-- 
2.25.1
```
...and removing `sysdeps/ia64/fpu/{s_round.S,s_roundf.S,s_roundl.S}`

2. adapting the symbol names for `sysdeps/ia64/fpu/{s_round.S,s_roundl.S}` accordingly. Interestingly the symbol names in `sysdeps/ia64/fpu/s_roundf.S` already use the `__` prefix. I don't know why that is and why it was not made for the above two, too.

**Comment by johnny-mnemonic** [comment]
Fixed in [toolchain-autobuilds/run-#182](https://github.com/johnny-mnemonic/toolchain-autobuilds/actions/runs/12189945068/job/34006078807#step:4:118).
